src/commands/dsyms/utils.ts

```diff
@@ -0,0 +1,112 @@
+import {exec} from 'child_process'
+import {promises} from 'fs'
+import glob from 'glob'
+import {tmpdir} from 'os'
+import path from 'path'
+import {promisify} from 'util'
+
+import {Payload} from './interfaces'
+
+import {buildPath} from '../../helpers/utils'
+
+const UUID_REGEX = '[0-9A-F]{8}-([0-9A-F]{4}-){3}[0-9A-F]{12}'
+
+const globAsync = promisify(glob)
+
+export const isZipFile = async (filepath: string) => {
+  try {
+    const stats = await promises.stat(filepath)
+
+    return stats.size !== 0 && path.extname(filepath) === '.zip'
+  } catch (error) {
+    // Log to console "file exists yet empty" ?
+    return false
+  }
+}
+
+export const getMatchingDSYMFiles = async (absoluteFolderPath: string): Promise<Payload[]> => {
+  const dSYMFiles = await globAsync(buildPath(absoluteFolderPath, '**/*.dSYM'))
+
+  return Promise.all(
+    dSYMFiles.map(async (dSYMPath) => {
+      const uuids = await dwarfdumpUUID(dSYMPath)
+
+      return {
+        path: dSYMPath,
+        type: 'ios_symbols',
+        uuids,
+      }
+    })
+  )
+}
+
+const dwarfdumpUUID = async (filePath: string) => {
+  const output = await execute(`dwarfdump --uuid ${filePath}`)
+
+  const uuids: string[] = []
+  output.stdout.split('\n').forEach((line: string) => {
+    const regexMatches = line.match(UUID_REGEX)
+    if (regexMatches && regexMatches.length > 0) {
+      uuids.push(regexMatches[0])
+    }
+  })
+
+  return uuids
+}
+
+const tmpFolder = buildPath(tmpdir(), 'datadog-ci', 'dsyms')
+
+export const zipToTmpDir = async (sourcePath: string, targetFilename: string): Promise<string> => {
+  await promises.mkdir(tmpFolder, {recursive: true})
+  const targetPath = buildPath(tmpFolder, targetFilename)
+  const sourceDir = path.dirname(sourcePath)
+  const sourceFile = path.basename(sourcePath)
+  // `zip -r foo.zip f1/f2/f3/foo.dSYM`
+  // this keeps f1/f2/f3 folders in foo.zip, we don't want this
+  // `cwd: sourceDir` is passed to avoid that
+  await execute(`zip -r ${targetPath} ${sourceFile}`, sourceDir)
```

Comment:
This does not seem to be platform independent and I suspect it would not work under Windows.
You could use `archiver` to do so in a platform agnostic way (cf. https://www.npmjs.com/package/archiver)
[As a side, it would also remove two occurrences of `execute` which is always a red flag security-wise)

Reply:
i agree, we can use a library in the upcoming releases to un/zip operations instead of spinning a new process.

but platform dependency shouldn't be a major concern for us as we target iOS devs.
i couldn't find `dwarfdump` for Windows and it is essential for `dsyms upload` command, we can't process dsym files without their UUID. 

as `dsyms upload` won't work in Windows, i added `if (os.platform() === 'darwin')` condition to the command

Reply:
i had to revert `if (os.platform() === 'darwin')` condition because our CI runs in a non-darwin platform.

now we have `dsyms upload` in all platforms, but it will crash in non-darwin platforms.
and i mock `dwarfdumpUUID` function in our unit tests.